File: prototypes/backend-javascript/backend-node-javascript-postgresql-sequelize/src/features/city/city-service.js
class CityService {
  constructor(repository) {
    this.repository = repository;
  }

  async getItems() {
    return await this.repository.getItems();
  }

  async getItemById(id) {
    return await this.repository.getItemById(id);
  }

  async createItem(city) {
    return await this.repository.createItem(city);
  }

  async updateItem(id, updatedData) {
    return await this.repository.updateItem(id, updatedData);
  }

  async deleteItem(id) {
    return await this.repository.deleteItem(id);
  }
}

export default CityService;

File: mcp/java/backend-java/src/main/java/com/ganatan/modules/city/CityService.java

package com.ganatan.modules.city;

import java.util.List;

public class CityService {

    private final CityRepository repository;

    public CityService(CityRepository  repository) {
        this.repository = repository;
    }

    public List<City> getItems() {
        return repository.getItems();
    }
}
